src/Sarif/RuleUtilities.cs

```diff
@@ -103,7 +104,7 @@ public static string NormalizeFormatId(string ruleId, string formatId)
                 throw new ArgumentNullException(nameof(formatId));
             }
 
-            if (!string.IsNullOrEmpty(ruleId) && formatId.StartsWith(ruleId + "_"))
+            if (!string.IsNullOrEmpty(ruleId) && string.Compare(formatId.Substring(0, ruleId.Length+1), ruleId+"_", StringComparison.Ordinal)==0)
```

Comment:
You don't need to introduce `string.Compare` just for the sake of including the comparison type. `StartsWith` has an overload that takes a comparison type, so you can write just `formatId.StartsWith(ruleId + "_", StringComparison.Ordinal)`.